tst/regression/CMakeLists.txt

```diff
@@ -1,36 +1,50 @@
 # AthenaPK - a performance portable block structured AMR MHD code
-# Copyright (c) 2020-2022, Athena Parthenon Collaboration. All rights reserved.
+# Copyright (c) 2020-2023, Athena Parthenon Collaboration. All rights reserved.
 # Licensed under the 3-Clause License (the "LICENSE");
 
+# Any external modules that are required by python can be added to REQUIRED_PYTHON_MODULES
+# list variable, before including TestSetup.cmake.
+list(APPEND REQUIRED_PYTHON_MODULES numpy unyt matplotlib h5py scipy)
+message(STATUS
+  "Checking for Python modules (${REQUIRED_PYTHON_MODULES}) required for regression tests.\n"
+  "   Check can be disabled by setting PARTHENON_ENABLE_PYTHON_MODULE_CHECK=OFF but then\n"
+  "   tests are not guaranteed to work anymore.")
+
 # import Parthenon setup_test_serial and setup_test_parallel
 include(${PROJECT_SOURCE_DIR}/external/parthenon/cmake/TestSetup.cmake)
 
-setup_test_serial("riemann_hydro" "--driver ${PROJECT_BINARY_DIR}/bin/athenaPK \
+function(setup_test_both dir arg test_labels)
+  setup_test_serial(${dir} "${arg}" "${test_labels}")
+  # just using 2 ranks for MPI tests
+  setup_test_parallel(2 ${dir} "${arg}" "${test_labels}")
+endfunction()
+
+setup_test_both("riemann_hydro" "--driver ${PROJECT_BINARY_DIR}/bin/athenaPK \
   --driver_input ${PROJECT_SOURCE_DIR}/inputs/sod.in --num_steps 39" "other")
 
-setup_test_serial("turbulence" "--driver ${PROJECT_BINARY_DIR}/bin/athenaPK \
+setup_test_both("turbulence" "--driver ${PROJECT_BINARY_DIR}/bin/athenaPK \
   --driver_input ${PROJECT_SOURCE_DIR}/inputs/turbulence.in --num_steps 1" "other")
 
-setup_test_serial("convergence" "--driver ${PROJECT_BINARY_DIR}/bin/athenaPK \
+setup_test_both("convergence" "--driver ${PROJECT_BINARY_DIR}/bin/athenaPK \
   --driver_input ${PROJECT_SOURCE_DIR}/inputs/linear_wave3d.in --num_steps 40" "convergence")
 
-setup_test_serial("mhd_convergence" "--driver ${PROJECT_BINARY_DIR}/bin/athenaPK \
+setup_test_both("mhd_convergence" "--driver ${PROJECT_BINARY_DIR}/bin/athenaPK \
   --driver_input ${PROJECT_SOURCE_DIR}/inputs/linear_wave3d.in --num_steps 48" "convergence")
 
 setup_test_serial("performance" "--driver ${PROJECT_BINARY_DIR}/bin/athenaPK \
```

Comment:
Why is this one serial while the rest are run both in serial and parallel?

Reply:
Because the CI machine has no NVLink between the GPU and is used for various other things simultaneously.
Thus, numbers for multi-GPU performance would not carry much significance.